lib/modem_info/modem_info.c

```diff
@@ -950,6 +956,42 @@ int modem_info_get_current_band(uint8_t *band)
 	return 0;
 }
 
+int modem_info_get_operator(char *buf, size_t len)
+{
+	if (buf == NULL || len < MODEM_INFO_MAX_SHORT_OP_NAME_SIZE) {
+		return -EINVAL;
+	}
+
+	char response[XMONITOR_CMD_MAX_RESPONSE_LEN];
+	int ret = nrf_modem_at_cmd(response, sizeof(response), AT_CMD_XMONITOR);
+
+	if (ret) {
+		return map_nrf_modem_at_scanf_error(ret);
+	}
+
+	// Return if no operator has been reported
+	if (strlen(response) < XMONITOR_CMD_MIN_RESPONSE_WITH_OPERATOR) {
+		return -ENOMSG;
+	}
+
+	// To get to the start of the short operator name, jump to the second comma
+	// then move forward two chars (skip comma + quote). To get to the end, jump to the comma
+	// after the start, then move back two chars (skip comma + quote).
+	// See below for an illustration:
+	//         %XMONITOR: 1,"LONG_OPERATOR_NAME","SHORT_OP_NAME",...
+	//                                            ^           ^
+	//                                            |           |
+	//                                          start        end
+	const char comma = ',';
+	char *start = strchr(response, comma) + 1;
+	start = strchr(start, comma) + 2;
+	char *end = strchr(start, comma) - 2;
```

Comment:
(no change needed) is the end not the next double quote? or is it safer to rely on the comma delimination?

Reply:
We could use the next double quote instead to jump -- to be consistent though, I like using only one character for delimitation. I don't think it is inherently safer; I'm just as reliant on a double quote being there since I currently skip over it, so I think they are both the same amount of safety if used for delimitation.